File: algorithm/HDU-ACM/HDU_ACM_C++/1222.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: devil_momo
////Nickname: Lucifer
////Run ID: 
////Submit time: 2008-07-06 07:52:23
////Compiler: Visual C++
//////////////////////////////////////////////////////
////Problem ID: 1222
////Problem Title: 
////Run result: Accept
////Run time:15MS
////Run memory:0KB
//////////////////System Comment End//////////////////
#include<iostream>
using namespace std;
int main()
{
	int a,b,r,m,n,cn;
	while(cin>>cn)
	{
		while(cn--)
		{
			cin>>m>>n;
			a=m,b=n;
			while(b)
			{
				r=a%b;
				a=b;
				b=r;
			}
			if(a==1)
				cout<<"NO\n";
			else
				cout<<"YES\n";
		}
	}
	return 0;
}

File: algorithm/HDU-ACM/ambition0109/1222.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-12 20:07:23
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1222
////Problem Title: 
////Run result: Accept
////Run time:109MS
////Run memory:6224KB
//////////////////System Comment End//////////////////
import java.util.Scanner;

public class Main {
	
	static int GCD(int a,int b){
		if(a<b){
			int t=a; a=b; b=t;
		}
		while(b!=0){
			int t=b;
			b=a%b;
			a=t;
		}
		return a;
	}
	
	public static void main(String args[]){
		Scanner in=new Scanner(System.in);
		int t=in.nextInt();
		while(t--!=0){
			int n=in.nextInt();
			int m=in.nextInt();
			System.out.println(GCD(n,m)==1?"NO":"YES");
		}
	}
}
